kustomize/plugin/giantswarm.io/v1/releasenoteurlannotationgenerator/releaseNoteURLAnnotationGenerator.go

```diff
@@ -0,0 +1,60 @@
+package main
```

Comment:
Lets try to compile a pro/con list to decide. WDYT?

Reasons to move into own repository:

- Keep this repository clean
- kustomize itself is already an external dependency
- Would allow to make the plugin more flexible (maybe useful for someone else)

Reasons **not** to move this into own repository:

- In its current state this isn't really useful for anyone else
- more complicated ci setup

Reply:
I would prefer to move but it does not happen _now now_
Would be interested in what other people think :D

Reply:
I think a separate repo for all custom kustomize plugins could make sense. 

This is a little crazy, but another option is to do edit CRDs at generation time rather than in CI avoiding kustomize completely. Here's an example of using the programmatic interface to `kustomize` rather than using it as an executable: https://github.com/giantswarm/apiextensions/pull/498/files This gives us CRDs as structs rather than writing them to YAML and dealing with the filesystem. I was planning to use this to validate some of our CRD comments and struct tags.

Reply:
Thank you for your feedback! I find your idea to use crd defaulting (you're referring to [this](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting) right?) really intriguing but are all control planes on 1.17 yet? Or doesn't it really matter because the open api defaulting isn't really useful here?

I admit, I don't have a clue where I would add this programmatic defaulting to apiextensions, so I'll probably go forward and move the kustomize plugin in this PR to an own repository.

I'm afraid I don'